proxy/src/registry/transaction.rs

```diff
@@ -65,10 +114,36 @@ pub enum Message {
 #[derive(Clone, Debug, Deserialize, Serialize)]
 #[serde(rename_all = "camelCase", tag = "type")]
 pub enum State {
-    /// [`Transaction`] has been applied to a block, carries the hash of the block.
-    Applied {
-        /// The hash of the block the transaction has been applied to.
-        block: registry::Hash,
+    /// [`Transaction`] has been applied to a block, carries the height of the block.
+    Confirmed {
+        /// The height of the block the transaction has been applied to.
+        block: protocol::BlockNumber,
+        /// Amount of progress made towards settlement. We assume height+6 to
+        /// be mathematically impossible to be reverted.
+        progress: u32,
+        /// Time when it was applied.
+        timestamp: Timestamp,
+    },
+
+    /// [`Transaction`] failed to be applied or processed.
+    // TODO(xla): Embbed original [`protocol::TransactionError`] and serialize it.
+    Failed {
+        /// Description of the error that occurred.
+        error: String,
+        /// Time when it failed.
+        timestamp: Timestamp,
```

Comment:
BlockNumber could be useful here too.

Reply:
The failure could be before the tx was applied, in this case we wouldn't have a block. We could make it a maybe.

Reply:
Yeah, could be a maybe. Though I'm not entirely sure what kind of failures can happen before the tx is applied. Would be good to figure that out because I think it's a very different type of failure.

Reply:
Failures before application to a block include codec, rpc and tx invalid errors. For reference `radicle_registry_client::Error`. (docs.rs doesn't have the latest for the crate so I can't link to it)

Reply:
Cool yeah, that's what I expected. This is most likely client errors then, eg. a bug in the client, or the client needs to be upgraded.